crates/sncast/src/helpers/scarb_utils.rs

```diff
@@ -196,48 +197,63 @@ pub fn get_scarb_metadata_with_deps(
     execute_scarb_metadata_command(&command)
 }
 
-#[must_use]
-pub fn verify_or_determine_scarb_manifest_path(
-    path_to_scarb_toml: &Option<Utf8PathBuf>,
-) -> Option<Utf8PathBuf> {
+pub fn ensure_scarb_manifest_path(path_to_scarb_toml: &Option<Utf8PathBuf>) -> Result<Utf8PathBuf> {
```

Comment:
I don't understand what this function does from the name.

Reply:
Carry over from https://github.com/foundry-rs/starknet-foundry/pull/1565#discussion_r1464663169

This function ensures valid path to manifest file. I agree that name may be a bit unfortunate, maybe I should change it to something like `get_scarb_manifest`?

Reply:
If it returns scarb manifest i think it would be a better name.

Reply:
it turned out there is a funtion named `get_scarb_manifest` (it uses scarb to get the manifest), so I have renamed `ensure_scarb_manifest_path` to `assert_manifest_path_exists`, as after some changes this function actually asserts its existence. (https://github.com/foundry-rs/starknet-foundry/pull/1569/commits/bcc6b0fac2cb70a8deddb51e6ba380034d1574d2)

Reply:
Updated docs + changelog (please check the docs especially, as I'm afraid I didn't stress enough the fact that only most top-level Scarb.toml holds the config https://github.com/foundry-rs/starknet-foundry/pull/1569/commits/40b52d3fb1ea6c6a83570824c676f582aa1d383e)

Moved flags to respective subcommands https://github.com/foundry-rs/starknet-foundry/pull/1569/commits/4791a4d94c28f71cd8e82dc90c9a358a283d8581